connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -2255,6 +2255,23 @@ class SparkConnectPlanner(val session: SparkSession) {
           .build()
         respBuilder.setExplain(explain)
 
+      case StreamingQueryCommand.CommandCase.EXCEPTION =>
+        val result = query.exception
+        result.foreach(e =>
+          respBuilder.getExceptionBuilder
+            .setExceptionMessage(SparkConnectService.extractErrorMessage(e)))
+
+      case StreamingQueryCommand.CommandCase.AWAIT_TERMINATION =>
+        if (command.getAwaitTermination.hasTimeoutMs) {
+          val terminated = query.awaitTermination(command.getAwaitTermination.getTimeoutMs)
+          respBuilder.getAwaitTerminationBuilder
+            .setTerminated(terminated)
+        } else {
+          query.awaitTermination()
```

Comment:
Hmm .. just to be extra clear, it will be disconnected when it reaches gRPC timeout .. am i correct?

Reply:
Right this is intended at this stage. @rangadi will push update regarding this I believe

Reply:
👌

Reply:
The client keep sending heart beat message to keep the RPC connection alive. 
That said, we would still need to improve handling of this. E.g. it should exit if client side disconnects.